nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/internal/HelmHostedIT.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2017-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.plugins.helm.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.sonatype.nexus.common.app.BaseUrlHolder;
+import org.sonatype.nexus.pax.exam.NexusPaxExamSupport;
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.Component;
+import org.sonatype.nexus.repository.storage.ComponentMaintenance;
+import org.sonatype.nexus.testsuite.testsupport.NexusITSupport;
+
+import org.apache.http.HttpResponse;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.ops4j.pax.exam.Configuration;
+import org.ops4j.pax.exam.Option;
+
+import static java.lang.String.format;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+public class HelmHostedIT
+    extends HelmITSupport
+{
+  private HelmClient client;
+
+  private Repository repository;
+
+  @Configuration
+  public static Option[] configureNexus() {
+    return NexusPaxExamSupport.options(
+        NexusITSupport.configureNexusBase(),
+        nexusFeature("org.sonatype.nexus.plugins", "nexus-repository-helm")
+    );
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    BaseUrlHolder.set(this.nexusUrl.toString());
+    repository = repos.createHelmHosted("helm-hosted-test");
+    client = createHelmClient(repository);
+    uploadPackages(MONGO_PKG_FILE_NAME_600_TGZ, MONGO_PKG_FILE_NAME_728_TGZ);
```

Comment:
Could we add an IT of the "empty" starting state of a new Hosted Helm repo?
I've seen issues in the past where clients didn't like the behavior of our empty repo (e.g. missing indexes, etc).

Reply:
It's not clear for me, what we'd like to test for the "empty" state ?

Reply:
I did a little testing to see if the helm client handled an empty NXRM Hosted Helm repo. 
Basically, I followed the steps for [Helm Testing](https://docs.sonatype.com/display/Nexus/Helm+Testing), but instead setup a Hosted repo.

My helm config had only this:

```
$ helm repo list
NAME       	URL                                          
local      	http://127.0.0.1:8879/charts                 
nexushosted	http://localhost:8081/repository/helm-hosted/
```

And a command to fetch a non-existent chart didn't seem to blowup:

```
$ helm fetch nexushosted/mongodb
Error: chart "mongodb" matching version "" not found in nexushosted index. (try 'helm repo update'). no chart name found
```

Also verified the suggested "helm repo update" command didn't get nasty as well:

```
$ helm repo update
Hang tight while we grab the latest from your chart repositories...
...Skip local chart repository
...Successfully got an update from the "nexushosted" chart repository
Update Complete.
```

So it looks like whatever NXRM currently returns is valid. It would be nice to have an IT that verified the return values when the Hosted repo is empty. Maybe like `testFetchPackage()`, but validating the case when the repo does not contain any charts?
e.g. 
resp.status code == 404
resp.getEntity().getContentType().getValue() == "text/html" 

Another thought is to test for fetching a non-existent chart (as that seems to return the same result). The empty case just seems more thorough, and I've seen that bite us before.